packages/uhk-web/src/app/store/reducers/user-configuration.ts

```diff
@@ -532,6 +532,33 @@ export const getMacroMap = (state: State): Map<number, Macro> => {
 };
 export const lastEditedKey = (state: State): LastEditedKey => state.lastEditedKey;
 
+export const getMacroUsage = (state: State): Array<object> => {
+    const macros = [];
+    const maps = state.userConfiguration.keymaps;
+    for (let mapsi = 0; mapsi < maps.length; mapsi++) {
+        const map = maps[mapsi];
+        for (let layersi = 0; layersi < map.layers.length; layersi++) {
+            const layer = map.layers[layersi];
+            for (let modulesi = 0; modulesi < layer.modules.length; modulesi++) {
+                const module = layer.modules[modulesi];
+                for (let keysi = 0; keysi < module.keyActions.length; keysi++) {
+                    const key = module.keyActions[keysi];
+                    if (key && key.hasOwnProperty('_macroId')) {
+                        macros.push({
```

Comment:
too complex object, you use it only the macroId else in the code. Use the `MacroMenuItem` structure that I mentioned earlier.